jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePane.kt

```diff
@@ -2,8 +2,101 @@
 // SPDX-License-Identifier: Apache-2.0
 package software.aws.toolkits.jetbrains.services.sqs.toolwindow
 
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.Message
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.services.sqs.Queue
+import software.aws.toolkits.resources.message
+import javax.swing.JLabel
 import javax.swing.JPanel
 
-class PollMessagePane {
+class PollMessagePane(
+    private val project: Project,
+    private val queue: Queue
+) : Disposable {
     lateinit var component: JPanel
+    lateinit var messagesAvailableLabel: JLabel
+    lateinit var tablePanel: SimpleToolWindowPanel
+
+    private val client: SqsClient = project.awsClient()
```

Comment:
This should not be creating a new client, but instead pass it along from the explorer so that it continues to operate even after they switch regions / creds

Reply:
Yes, that would be optimal. I do have a concern though because from my understanding about service classes like `SqsWindow`, I am not allowed to pass any arguments other than Project. If that's the case, I would not be able to pass it along from explorer, but instead, I can move this line to SqsWindow so I wouldn't have to create a new client every time `PollMessagePane` is called. Please correct me if I am wrong.